homeassistant/components/loqed/sensor.py

```diff
@@ -12,31 +14,83 @@
     PERCENTAGE,
     SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
     EntityCategory,
+    UnitOfElectricPotential,
 )
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
 
-from .const import DOMAIN
+from .const import BATTERY_TYPES, DOMAIN, EVENT_TYPES
 from .coordinator import LoqedDataCoordinator, StatusMessage
 from .entity import LoqedEntity
 
-SENSORS: Final[tuple[SensorEntityDescription, ...]] = (
-    SensorEntityDescription(
+
+@dataclass(kw_only=True)
+class LoqedSensorEntityDescription(SensorEntityDescription):
+    """Dataclass describing LOQED sensor entities."""
+
+    value_fn: Callable[[StatusMessage], StateType]
+
+
+SENSORS: Final[tuple[LoqedSensorEntityDescription, ...]] = (
+    LoqedSensorEntityDescription(
+        key="last_event",
+        value_fn=lambda data: data.last_event if data.last_event != "" else None,  # type: ignore[attr-defined]
```

Comment:
```suggestion
        value_fn=lambda data: data.last_event if data.last_event else None,  # type: ignore[attr-defined]
```

Reply:
`data` is a dict, right?

`StatusMessage` is defined as a `TypedDict`, something is off here requiring to ignore typing because of that?

Reply:
For some reason mypy triggers warning `StatusMessage has not attribute 'last_event'`.

Reply:
Correct, as the TypedDict wasn't extended, also, it is a dict, but you are accessing it as an object. Tests pass, as the tests seems to have a mismatch in the mock (as the mock created objects, but the library seems to return dictionaries?)

Something is fishy here :D

Reply:
But it is Typed as a dictionary:

<https://github.com/home-assistant/core/blob/9741380cc09592b74eb34da11f66e48f4706a05f/homeassistant/components/loqed/coordinator.py#L51>

Reply:
@frenck I can only speculate what was initial idea behind this type setup. Probably `StatusMessage` was meant to be used as interface? 🤷🏻

Reply:
@mikewoudenberg Would you mind to comment this issue pls?

Reply:
The data coordinator is initialised with with data coming back from the bridge, which is a StatusMessage. The lock property on the coordinator is a loqed.Lock. Hope that sheds some light

Reply:
Not really? Is it an object or a dictionary?

Reply:
The sensors read data from the `lock` property on the datacoordinator so the return type of the data property on the LoqedSensor class should be `loqed.Lock`, which is an object

Reply:
@mikewoudenberg thanks. I've updated PR :)

@frenck pls check again. It should be solved now.